src/jwplayer/players/shared/JwPlayerWrapperWithStrategyRules.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useContext } from 'react';
+import { JWPauseEvent, JWPlacementApi, JWPlayerApi, JWPlayEvent, JWPPlacementReadyResponse } from 'jwplayer/types';
+import { PlayerContext } from 'jwplayer/players/shared/PlayerContext';
+import { JwPlayerWrapperProps } from 'jwplayer/types';
+import { jwPlayerPlaybackTracker, triggerVideoMetric } from 'jwplayer/utils/videoTracking';
+import { recordVideoEvent, VIDEO_RECORD_EVENTS } from 'jwplayer/utils/videoTimingEvents';
+import JWEvents from 'jwplayer/players/shared/JWEvents';
+import addBaseTrackingEvents from 'jwplayer/players/shared/addBaseTrackingEvents';
+import getSponsoredVideos from 'utils/getSponsoredVideos';
+
+interface WindowJWPlayer extends Window {
+	jwplayer?: JWPlayerApi;
+	jwplacements?: JWPlacementApi;
+	sponsoredVideos?: string[];
+}
+
+declare let window: WindowJWPlayer;
+
+const JwPlayerWrapperWithStrategyRules: React.FC<JwPlayerWrapperProps> = ({
+	config,
+	onReady,
+	getDismissed = () => false,
+	onComplete,
+	className,
+	shouldLoadSponsoredContentList = true,
+	jwPlayerContainerEmbedId = 'featured-video__player',
+	vastUrl,
+}) => {
+	const { setPlayer, setConfig } = useContext(PlayerContext);
+	const videoIndexRef = React.useRef(0);
+	const strategyRulesPlacementId = '21rL5wJF';
+	const recommendationPlaylistId = 'FOhaD53w';
+
+	useEffect(() => {
```

Comment:
It could be shared between both JwPlayerWrappers. Not sure as a part of this PR as this is only PoC 🤷‍♂️

Reply:
I tried and the new hook seems to work fine. Take a look please and let me know if there are no red flags 😊 https://github.com/Wikia/jwplayer-fandom/pull/195/commits/33dcedba3c88af4354192076058d515aafbdae32

Reply:
LGTM